### ISSUE TITLE
DDP server ping message blocked when publishing lots of data, triggers infinite-reconnect.

### ISSUE THREAD

**Issue by rogerrathers1** [meteor/meteor#5432]
@gchpaco fixed #4298 which was released in Meteor 1.2.  Because ping/pong messages can be blocked by other messages, all messages are now considered pings by the heartbeat code.

Long running publishes have a bug in Meteor 1.2.0.2. If the server publishes many records, the client sees all messages as a 'pong' and so knows the connection is alive. Because of that it doesn't send a ping request. The server sees that the client is silent and so it sends a ping request. The ping request is queued behind the publish messages and so the connection times out and the server closes the connection.

A simple bugfix is to have the client send periodic pong messages. 

The following repository demonstrates the bug and the bugfix: https://github.com/rogerrathers1/server_ddp_ping_blocked_infinite_reconnect

**Comment by gchpaco** [comment]
Yeah, that's a problem.  In-band health signaling is rife with this.  I'll see what I can do.
